text/0000-slice-copy-fill.md

```diff
@@ -0,0 +1,99 @@
+- Feature Name: std::slice::copy, slice::fill
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Safe `memcpy` from one slice to another of the same type and length, and a safe
+`memset` of a slice of type `T: Copy`.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the only way to quickly copy from one non-`u8` slice to another is to
+use a loop, or unsafe methods like `std::ptr::copy_nonoverlapping`. This allows
+us to guarantee a `memcpy` for `Copy` types, and is safe. The only way to
+`memset` a slice, currently, is a loop, and we should expose a method to allow
+people to do this. This also completely gets rid of the point of
+`std::slice::bytes`, which means we can remove this deprecated and useless
+module.
+
+# Detailed design
+[design]: #detailed-design
+
+Add two methods to Primitive Type `slice`.
+
+```rust
+impl<T> [T] where T: Copy {
+    pub fn fill(&mut self, value: T);
+    pub fn copy_from(&mut self, src: &[T]);
+}
+```
+
+`fill` loops through slice, setting each member to value. This will lower to a
+memset in all possible cases. It is defined behavior to call `fill` on a slice
+which has uninitialized members, and `self` is guaranteed to be fully filled
+afterwards.
+
+`copy` panics if `src.len() != self.len()`, then `memcpy`s the members into 
+`self` from `src`. Calling `copy_from` is semantically equivalent to a `memcpy`;
+`self` can have uninitialized members, and `self` is guaranteed to be fully filled
+afterwards. This means, for example, that the following is fully defined:
+
+```rust
+let s1: [u8; 16] = unsafe { std::mem::uninitialized() };
+let s2: [u8; 16] = unsafe { std::mem::uninitialized() };
+s1.fill(42);
+s2.copy_from(&s1);
+println!("{}", s2);
+```
+
+And the program will print 16 '8's.
```

Comment:
I think @nagisa meant that it would be quite surprising to see such program printing `8` instead of `42` (in fact it would definitely make me think that something was not initialised properly :P ).